lib/internal/modules/cjs/loader.js

```diff
@@ -567,7 +570,16 @@ function tryModuleLoad(module, filename) {
 }
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (NativeModule.nonInternalExists(request)) {
+  if (request.startsWith('@nodejs/')) {
+    request = request.slice(8);
+    if (!namespaceWhitelist.has(request)) {
+      throw new ERR_INVALID_NODE_BUILTIN(request);
```

Comment:
Can you please add a test to verify this case? I haven't found one in this PR. This is a very important feature that prevents misusage.